src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+          <HelpItem
+            helpText={th("singleLogoutServiceUrl")}
+            forLabel={t("singleLogoutServiceUrl")}
+            forID="single-logout-service-url"
+          />
+        }
+        fieldId="single-logout-service-url"
+        validated={
+          errors.config?.singleLogoutServiceUrl
+            ? ValidatedOptions.error
+            : ValidatedOptions.default
+        }
+        helperTextInvalid={t("common:required")}
+      >
+        <TextInput
+          type="text"
+          id="single-logout-service-url"
+          name="config.singleLogoutServiceUrl"
+          isReadOnly={readOnly}
+        />
+      </FormGroup>
+
+      <SwitchField
+        field="config.backchannelSupported"
+        label="backchannelLogout"
+        isReadOnly={readOnly}
+      />
+
+      <FormGroup
+        label={t("nameIdPolicyFormat")}
+        labelIcon={
+          <HelpItem
+            helpText={th("nameIdPolicyFormat")}
+            forLabel={t("nameIdPolicyFormat")}
+            forID="kc-nameIdPolicyFormat"
+          />
+        }
+        fieldId="kc-nameIdPolicyFormat"
+        helperTextInvalid={t("common:required")}
+      >
+        <Controller
+          name="config.nameIDPolicyFormat"
+          defaultValue={t("persistent")}
+          control={control}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-nameIdPolicyFormat"
+              onToggle={() =>
+                setNamedPolicyDropdownOpen(!namedPolicyDropdownOpen)
+              }
+              isOpen={namedPolicyDropdownOpen}
+              onSelect={(_, value) => {
+                onChange(value as string);
+                setNamedPolicyDropdownOpen(false);
+              }}
+              selections={value}
+              variant={SelectVariant.single}
+            >
+              <SelectOption
+                key={0}
+                data-testid="persistent-option"
+                value={t("persistent")}
+                isPlaceholder
+              />
+              <SelectOption
+                key={1}
+                data-testid="transient-option"
+                value={t("transient")}
+              />
+              <SelectOption
+                key={2}
+                data-testid="email-option"
+                value={t("email")}
+              />
+              <SelectOption
+                key={3}
+                data-testid="kerberos-option"
+                value={t("kerberos")}
+              />
+              <SelectOption
+                key={4}
+                data-testid="x509-option"
+                value={t("x509")}
+              />
+              <SelectOption
+                key={5}
+                data-testid="windowsDomainQN-option"
+                value={t("windowsDomainQN")}
+              />
+              <SelectOption
+                key={6}
+                data-testid="unspecified-option"
+                value={t("unspecified")}
+              />
```

Comment:
I don't think it's needed to give these `SelectOption` a `key` prop, as they are statically populated.
```suggestion
              <SelectOption
                data-testid="persistent-option"
                value={t("persistent")}
                isPlaceholder
              />
              <SelectOption
                data-testid="transient-option"
                value={t("transient")}
              />
              <SelectOption
                data-testid="email-option"
                value={t("email")}
              />
              <SelectOption
                data-testid="kerberos-option"
                value={t("kerberos")}
              />
              <SelectOption
                data-testid="x509-option"
                value={t("x509")}
              />
              <SelectOption
                data-testid="windowsDomainQN-option"
                value={t("windowsDomainQN")}
              />
              <SelectOption
                data-testid="unspecified-option"
                value={t("unspecified")}
              />
```

Reply:
interesting, i thought it would throw a missing key prop without these, but it doesn't. i removed them, ty.